src/azure-cli/azure/cli/command_modules/appservice/commands.py

```diff
@@ -340,3 +344,12 @@ def load_command_table(self, _):
         g.custom_command('add', 'add_webapp_access_restriction')
         g.custom_command('remove', 'remove_webapp_access_restriction')
         g.custom_command('set', 'set_webapp_access_restriction')
+
+    with self.command_group('appservice ase', custom_command_type=appservice_environment, is_preview=True) as g:
+        g.custom_command('list', 'list_appserviceenvironments')
+        g.custom_command('list-addresses', 'list_appserviceenvironment_addresses')
+        g.custom_command('list-plans', 'list_appserviceenvironment_plans')
+        g.custom_command('show', 'show_appserviceenvironment')
+        g.custom_command('create', 'create_appserviceenvironment_arm', supports_no_wait=True)
```

Comment:
is there any validation we need to add here? Like validating if the name is available & is valid?

Reply:
In both cases, it will fail with a descriptive error message, so I am not sure there is an urgent need to add validation.

Reply:
Not quite what i meant  - we usually validate before we call the ARM template or API for the creation process we have a 
checknameavailability API - please check with Mitch if you have more questions (we use this in APP create currently) where we validate with the resource is available before we try the create operation.

Reply:
Validating if the name is available now.